app/models/bulk_download.rb

```diff
@@ -23,6 +23,47 @@ class BulkDownload < ApplicationRecord
   attr_accessor :params
 
   validates :status, presence: true, inclusion: { in: [STATUS_WAITING, STATUS_RUNNING, STATUS_ERROR, STATUS_SUCCESS] }
+  validate :params_checks
+
+  def params_checks
+    set_params
```

Comment:
Params is typically to be used in a controller.... Doesnt this assume that it comes from a query?
What if we want to use BulkDownloads outside of an api call? Would this work?

Reply:
I ended up re-writing how `params` is handled. Now there's a reader method that tries to load `params` from `params_json` if `params` isn't defined.

The previous approach was bad because you had to remember to call `set_params`. Now it happens transparently.